wgpu-hal/Cargo.toml

```diff
@@ -66,6 +67,11 @@ egl = { package = "khronos-egl", version = "4.1", features = ["dynamic"], option
 #Note: it's only unused on Apple platforms
 libloading = { version = "0.7", optional = true }
 
+[target.'cfg(target_os = "emscripten")'.dependencies]
+egl = { package = "khronos-egl", version = "4.1", features = ["static", "no-pkg-config"] }
```

Comment:
that's a bit weird: "static" actually implies "pkg-config":
https://github.com/timothee-haudebourg/khronos-egl/blob/76d7d917f1c5b40061596653a16f0340683b5e0b/Cargo.toml#L15

Reply:
This is from [official documentation](https://github.com/timothee-haudebourg/khronos-egl)
### Static linking

You must enable static linking using the `static` feature in your `Cargo.toml`:
```toml
khronos-egl = { version = ..., features = ["static"] }
```

This will add a dependency to the [`pkg-config`](https://crates.io/crates/pkg-config) crate,
necessary to find the EGL library at compile time.

If you wish to disable linking EGL in this crate, and provide linking in
your crate instead, enable the `no-pkg-config` feature.
```toml
khronos-egl = {version = ..., features = ["static", "no-pkg-config"]}
```

--

We use "no-pkg-config" because emscripten provides own implementation of EGL

Reply:
thanks! that's some perverted stuff here, but we are complying. Your code is good then